src/main/java/org/isf/anamnesis/gui/PatientHistoryEdit.java

```diff
@@ -0,0 +1,2629 @@
+		}
+		return jLabelPhyMenopauseYears;
+	}
+
+	private JButton getJButtonSave() {
+		if (jButtonSave == null) {
+			jButtonSave = new JButton();
+			jButtonSave.setAction(getActionSavePatientHistory());
+		}
+		return jButtonSave;
+	}
+
+//	private JButton getJButtonExamination() {
+//		if (jButtonExamination == null) {
+//			jButtonExamination = new JButton();
+//			jButtonExamination.setAction(getActionInsertExamination());
+//		}
+//		return jButtonExamination;
+//	}
+
+	private JButton getJButtonCancel() {
+		if (jButtonCancel == null) {
+			jButtonCancel = new JButton(MessageBundle.getMessage("angal.anamnesis.cancel.txt")); //$NON-NLS-1$
+			jButtonCancel.setMnemonic(KeyEvent.VK_C);
+			jButtonCancel.addActionListener(new ActionListener() {
+				public void actionPerformed(ActionEvent e) {
+					dispose();
+				}
+			});
+		}
+		return jButtonCancel;
+	}
+
+	class ActionSavePatientHistory extends AbstractAction {
+		/**
+		 *
+		 */
+		private static final long serialVersionUID = 1L;
+
+		public ActionSavePatientHistory() {
+
+			putValue(NAME, MessageBundle.getMessage("angal.anamnesis.save.txt")); //$NON-NLS-1$
+			putValue(MNEMONIC_KEY, KeyEvent.VK_S);
+			putValue(SHORT_DESCRIPTION, MessageBundle.getMessage("angal.anamnesis.tooltip.savethepatienthistory.txt")); //$NON-NLS-1$
+		}
+
+		public void actionPerformed(ActionEvent e) {
+
+			updateModelFromGUI();
+			if (storeData) {
+				patientHistoryManager.saveOrUpdate(path);
+			}
+			dispose();
+		}
+	}
+
+	private Action getActionSavePatientHistory() {
+		if (actionSavePatientHistory == null) {
+			actionSavePatientHistory = new ActionSavePatientHistory();
+		}
+		return actionSavePatientHistory;
+	}
+
+	private class SwingActionInsertExamination extends AbstractAction {
+		/**
+		 *
+		 */
+		private static final long serialVersionUID = 1L;
+
+		public SwingActionInsertExamination() {
+			putValue(NAME, MessageBundle.getMessage("angal.anamnesis.examination.txt")); //$NON-NLS-1$
+			putValue(MNEMONIC_KEY, KeyEvent.VK_E);
+			putValue(SHORT_DESCRIPTION, MessageBundle.getMessage("angal.anamnesis.tooltip.insertpatientexamination.txt")); //$NON-NLS-1$
+		}
+		public void actionPerformed(ActionEvent e) {
+
+//			PatientExamination patex;
+//			ExaminationOperations exaServ = new ExaminationOperations();
+
+//			PatientExamination lastPatex = exaServ.getLastByPatID(pat.getCode());
+//			if (lastPatex != null) {
+//				patex = exaServ.getFromLastPatientExamination(lastPatex);
+//			} else {
+//				patex = exaServ.getDefaultPatientExamination(pat.getCode());
+//			}
+//
+//			GenderPatientExamination gPatex = new GenderPatientExamination(patex, pat.getSex() == 'M');
+//			PatientExaminationEdit dialog = new PatientExaminationEdit(PatientHistoryEdit.this, gPatex);
+//			dialog.setDefaultCloseOperation(JDialog.DISPOSE_ON_CLOSE);
+//			dialog.pack();
+//			dialog.setLocationRelativeTo(null);
+//			dialog.setVisible(true);
+		}
```

Comment:
So this action does nothing?

Reply:
obsolete code. I removed also another piece of code. Please make a fast check of the GUI. I have made some tests and seems that it is all ok.